crates/core/src/db/datastore/locking_tx_datastore/committed_state.rs

```diff
@@ -644,6 +644,39 @@ impl CommittedState {
         let index = table.indexes.get(col_list)?;
         Some(&index.key_type)
     }
+
+    pub(super) fn report_data_size(&self, database_identity: Identity) {
```

Comment:
If this causes a performance issue because of time spent in `with_label_values`, we could store these metric handles in the `Table` and/or `ExecutionContext`, so reporting values would just be an atomic number operation.

Reply:
I've created a ticket to track this.